lib/accessmanager.js

```diff
@@ -159,16 +168,24 @@ AccessManager.prototype.updateToken = function updateToken(newToken) {
       self.emit('tokenUpdated', self);
     });
 
-    // Set expires, isExpired, and tokenExpired timer.
+    // Set expires, isExpired, tokenExpired and tokenWillExpire timers.
     self._expires = new Date(payload.exp * 1000);
     self._isExpired = Date.now() >= self.expires;
+
     clearTimeout(self._tokenExpiredTimer);
     self._tokenExpiredTimer = setTimeout(function tokenExpiredTimer() {
       self._isExpired = true;
       self._tokenExpiredTimer = null;
       self.emit('tokenExpired', self);
     }, self.expires - Date.now());
 
+    clearTimeout(self._tokenWillExpireTimer);
+    self._tokenWillExpireTimer = setTimeout(function tokenWillExpireTimer() {
+      self._tokenWillExpireTimer = null;
+      self.emit('tokenWillExpire', self);
+    }, self.expires - Date.now() - 3 * 60 * 1000); // 3 minutes before expiration
+    // Negative timeout for setTimeout is fine and means "almost now"
+
```

Comment:
Will you please document the event at the end of the file, like the other events?